### PR TITLE
First steps in Pyret parsing working

### PR DESCRIPTION
**Pull request by prestontw**
There are still known issues with pretty-printing, but this at least gets Pyret parsing to the point where `new-pyret-editor.html` opens and parses.

### CODE REVIEW

File: src/CodeMirrorBlocks.js [hunk 1]
```diff
@@ -1,19 +1,20 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import ToggleEditor from './ui/ToggleEditor';
-import wescheme from '../src/languages/wescheme';
+import pyret from '../src/languages/pyret';
```
Comment (schanzer) [inline src/CodeMirrorBlocks.js]: We should include _both_ languages here -- no reason for the constructor to only know about one of them!

File: src/components/NodeEditable.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ class NodeEditable extends Component {
         },
         e => {
           const errorText = SHARED.parser.getExceptionMessage(e);
-          say(errorText);
```
Comment (schanzer) [inline src/components/NodeEditable.js]: `say` logs to the console anyway

File: src/languages/pyret/ast.js [hunk 3]
```diff
@@ -101,15 +106,19 @@ export class Func extends ASTNode {
   }
 
   pretty() {
-    return P.lambdaLikeSexpr(this.id, this.args, this.body);
+    return P.horzArray([P.txt("fun "), this.name, P.txt("("), P.horzArray(this.args.map(p => p.pretty())), P.txt(")"), this.body]);
   }
 
   render(props) {
+    // TODO: uncommenting this expression leads to using an object instead of a react element
+    let args = /* this.args[0].reactElement() */ undefined;
```
Comment (schanzer) [inline src/languages/pyret/ast.js]: @justinpombrio ?

File: src/ui/ToggleEditor.js [hunk 4]
```diff
@@ -153,7 +154,7 @@ export default class ToggleEditor extends React.Component {
       } catch (err) {
         // TODO(Justin): properly deal with parse errors
         let msg = SHARED.parser.getExceptionMessage(err);
-        say(msg);
```
Comment (schanzer) [inline src/ui/ToggleEditor.js]: `say` already logs.

File: src/CodeMirrorBlocks.js (hunk 1 above)
Comment (prestontw) [reply src/CodeMirrorBlocks.js]: This is for the purpose of the default constructor. If we imported both, one of them would be unused.

Comment (prestontw) [reply src/CodeMirrorBlocks.js]: But I agree that we should remove this or document this better when other languages start using CMB.

File: src/components/NodeEditable.js (hunk 2 above)
Comment (prestontw) [reply src/components/NodeEditable.js]: But it also reads out loud the entirety of the error message! That's an awful user experience.

File: src/ui/ToggleEditor.js (hunk 4 above)
Comment (prestontw) [reply src/ui/ToggleEditor.js]: And reads the entirety of the error message out loud.

File: src/languages/pyret/ast.js [hunk 5]
```diff
@@ -73,7 +74,11 @@ export class Bind extends ASTNode {
   }
 
   pretty() {
-    return P.standardSexpr(this.id, [this.ann]);
+    console.log(this.id);
+    if (this.ann.type != "a-blank")
+      return P.horzArray([this.id.value, P.txt(" :: "), this.ann]);
+    else
+      return P.horzArray([this.id.value]);
```
Comment (justinpombrio) [inline src/languages/pyret/ast.js]: should use `txt`

File: src/languages/pyret/ast.js [hunk 6]
```diff
@@ -101,15 +106,19 @@ export class Func extends ASTNode {
   }
 
   pretty() {
-    return P.lambdaLikeSexpr(this.id, this.args, this.body);
+    return P.horzArray([P.txt("fun "), this.name, P.txt("("), P.horzArray(this.args.map(p => p.pretty())), P.txt(")"), this.body]);
```
Comment (justinpombrio) [inline src/languages/pyret/ast.js]: Should (1) have print in two different ways, depending on whether it fits on one line, and (2) make a helper function to display a comma-separated list for the args.

File: src/languages/pyret/ast.js (hunk 3 above)
Comment (justinpombrio) [reply src/languages/pyret/ast.js]: That line is doing the right thing type-wise. Something else is going on: maybe `args` is not an array of `Bind`?

Comment (justinpombrio) [reply src/languages/pyret/ast.js]: (Preston just confirmed: issue's in bind.id)

File: src/languages/pyret/ast.js (hunk 5 above)
Comment (prestontw) [reply src/languages/pyret/ast.js]: updated in new file, so resolving

File: src/languages/pyret/ast.js (hunk 6 above)
Comment (prestontw) [reply src/languages/pyret/ast.js]: updated in new file, so resolving

**Pull request merged**